front_end/WorkoutWise/src/pages/RegisterPage.jsx

```diff
@@ -1,7 +1,50 @@
+import { useState, useContext, useEffect } from "react";
+import { userContext } from "../App";
+import { api } from "../utilities";
+import { useNavigate } from "react-router-dom";
+
 export const RegisterPage = () => {
+    const [userName, setUserName] = useState("");
+    const [password, setPassword] = useState("");
+    const { setUser } = useContext(userContext);
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        console.log("username:  ",userName);
+        console.log("password:  ", password);
+    }, [userName, password]);
+
+    const signUp = async (e) => {
+        e.preventDefault();
+        let response = await api.post("users/register/",{
+            email: userName,
+            password: password,
+        });
+        console.log(response)
+        let user = response.data.user;
+        let token = response.data.token;
+        setUser(user);
+        localStorage.setItem("token", token);
+        navigate("/workouts");
+    };
+
     return(
-        <>
+        <form onSubmit={(e) => signUp(e)}>
             <h1>RegisterPage</h1>
-        </>
+            <h5>Register</h5>
+            <input 
+                type="email"
+                placeholder="email"
+                value={userName}
```

Comment:
Looks like theres a lot of RegisterPage and LogInPage are sharing a lot of the same code but just treat the onSubmit function a bit different. Please try to isolate these components and enforce the Don't Repeat Yourself principle